src/main/java/heronarts/lx/color/GradientUtils.java

```diff
@@ -193,25 +197,49 @@ public enum BlendMode {
     }),
 
     HSV((c1, c2, lerp) -> {
+      float hue1 = c1.hue;
+      float hue2 = c2.hue;
+      float sat1 = c1.saturation;
+      float sat2 = c2.saturation;
+      float bright1 = c1.brightness;
+      float bright2 = c2.brightness;
+      if (c1.isBlack()) {
+        hue1 = hue2;
+        sat1 = 0;
+      } else if (c2.isBlack()) {
+        hue2 = hue1;
+        sat2 = 0;
+      }
       return LXColor.hsb(
-        LXUtils.lerpf(c1.hue, c2.hue, lerp),
-        LXUtils.lerpf(c1.saturation, c2.saturation, lerp),
-        LXUtils.lerpf(c1.brightness, c2.brightness, lerp)
+        LXUtils.lerpf(hue1, hue2, lerp),
+        LXUtils.lerpf(sat1, sat2, lerp),
+        LXUtils.lerpf(bright1, bright2, lerp)
       );
     }),
 
     HSV2((c1, c2, lerp) -> {
       float hue1 = c1.hue;
       float hue2 = c2.hue;
+      float sat1 = c1.saturation;
+      float sat2 = c2.saturation;
+      float bright1 = c1.brightness;
```

Comment:
no need to pull these brightness values out to extra stack variables that are never modified...

Reply:
Done